routes/login/login.controller.js

```diff
@@ -19,9 +23,11 @@ const postCode = (req, res) => {
   let accessCode = req.body.code || null
   req.session = accessCode ? { code: accessCode } : null
 
-  if (accessCode && redirect) {
+  const errors = validationResult(req)
+
+  if(!errors.isEmpty()) {
+    res.status(422).render('login/code', { title: 'Enter access code', data: req.session || {}, errors: errorArray2ErrorObject(errors) })
+  } else if (accessCode && redirect) {
     return res.redirect(redirect)
   }
-
-  res.status(422).render('login/code', { title: 'Enter access code', data: req.session || {} })
```

Comment:
Hmm, this method doesn't return anything by default. 

I think it should be changed so that there's one `if`, and then falls through to some other behaviour.

But for now it's okay: we can validate `redirect` is there in some middleware and then we don't have to run that check in here.